src/overlays/actors/ovl_En_Kaizoku/z_en_kaizoku.c

```diff
@@ -1418,76 +1456,76 @@ void func_80B88964(EnKaizoku* this, PlayState* play) {
         this->swordState = -1;
     }
 
-    this->unk_2D8 = 0;
-    if ((curFrame >= this->animEndFrame) && (this->unk_2D0 < 2)) {
+    this->animationsDisabled = false;
+    if ((curFrame >= this->animEndFrame) && (this->spinAttackState < 2)) {
         if (!Actor_IsFacingPlayer(&this->picto.actor, 0x1554)) {
-            func_80B872A4(this);
-            this->unk_2B2 = Rand_ZeroOne() * 5.0f + 5.0f;
+            EnKaizoku_SetupReady(this);
+            this->combatTimer = Rand_ZeroOne() * 5.0f + 5.0f;
             this->lookTimer = 46;
             return;
         }
 
-        if (this->unk_2D0 != 0) {
-            func_80B87E28(this);
+        if (this->spinAttackState != 0) {
+            EnKaizoku_SetupRollBack(this);
             return;
         }
 
         if ((Rand_ZeroOne() > 0.7f) || (this->picto.actor.xzDistToPlayer >= 120.0f)) {
-            func_80B872A4(this);
-            this->unk_2B2 = Rand_ZeroOne() * 5.0f + 5.0f;
+            EnKaizoku_SetupReady(this);
+            this->combatTimer = Rand_ZeroOne() * 5.0f + 5.0f;
             return;
         }
 
         this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
         if (BREG(12) * 0.1f + 0.01f * 40.0f < Rand_ZeroOne()) {
-            func_80B87900(this);
+            // dev seemed unsatisfied with randomized behavior
+            EnKaizoku_SetupBlock(this); // in OOT this was sidestep
             return;
         }
 
-        v1 = ABS_ALT(player->actor.shape.rot.y - this->picto.actor.shape.rot.y);
-        if (v1 <= 0x2710) {
-
-            a0 = ABS_ALT(this->picto.actor.yawTowardsPlayer - this->picto.actor.shape.rot.y);
-            if (a0 > 0x4000) {
+        angleToPlayer = ABS_ALT(player->actor.shape.rot.y - this->picto.actor.shape.rot.y);
+        if (angleToPlayer <= 0x2710) {
+            yawDiff = ABS_ALT(this->picto.actor.yawTowardsPlayer - this->picto.actor.shape.rot.y);
+            if (yawDiff > 0x4000) {
                 this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
-                func_80B88CD8(this);
+                EnKaizoku_SetupCircle(this);
             } else {
-                func_80B85A00(this, play, true);
+                EnKaizoku_ReactToPlayer(this, play, true);
             }
         } else {
-            func_80B88CD8(this);
+            EnKaizoku_SetupCircle(this);
         }
     }
 }
 
-void func_80B88CD8(EnKaizoku* this) {
-    EnKaizoku_ChangeAnim(this, EN_KAIZOKU_ANIM_3);
+void EnKaizoku_SetupCircle(EnKaizoku* this) {
+    EnKaizoku_ChangeAnim(this, KAIZOKU_ANIM_SIDESTEP);
     this->picto.actor.speed = Rand_CenteredFloat(12.0f);
     this->skelAnime.playSpeed = 1.0f;
     this->picto.actor.world.rot.y = this->picto.actor.shape.rot.y;
-    this->unk_2B2 = Rand_ZeroOne() * 30.0f + 30.0f;
-    this->action = KAIZOKU_ACTION_3;
-    this->actionFunc = func_80B88D6C;
-    this->unk_2E0 = 0.0f;
+    this->combatTimer = Rand_ZeroOne() * 30.0f + 30.0f;
+    this->action = KAIZOKU_ACTION_SIDESTEP;
```

Comment:
Shouldn't the action be `KAIZOKU_ACTION_CIRCLE` then? I'm fine with the animation being `SIDESTEP` (since there's nothing about the animation that suggests that it's being used to circle the player), but for the action, it makes sense to name it similarly to the action function

Reply:
For some reason I thought cirlc was already used, but yeah it looks like CIRCLE is correct